### PR TITLE
Specs/channels

### PR DESCRIPTION
**Pull request by marto83**
This is just for the Ably.Channels functionality. I'm going to send you the individual channel specs pull request later.

### CODE REVIEW

File: src/IO.Ably.Tests/Rest/ChannelSpecs.cs [hunk 1]
```diff
@@ -44,6 +45,43 @@ public void ShouldBeAbleToCheckIsAChannelExists()
...
+            {
+                var channel = _client.Channels.Get("new");
+                var secondTime = _client.Channels.Get("new");
+                channel.Should().BeSameAs(secondTime);
+            }
+
+            [Fact]
+            [Trait("spec", "RSN3a")]
+            [Trait("spec", "RSN3b")]
+            public void WithProvidedChannelOptions_ShouldSetOptionsOnChannel()
```
Comment (mattheworiordan) [inline src/IO.Ably.Tests/Rest/ChannelSpecs.cs]: You my want to test what happens when you access a channel with options that was previously instanced without options, according to the spec it then sets the options on that channel

File: src/IO.Ably.Tests/Rest/ChannelSpecs.cs [hunk 2]
```diff
@@ -80,6 +80,17 @@ public void WithProvidedChannelOptions_ShouldSetOptionsOnChannel()
                 var channel = _client.Channels.Get("test", options);
                 (channel as RestChannel).Options.ShouldBeEquivalentTo(options);
             }
+
+            [Fact]
+            [Trait("spec", "RSN3c")]
+            public void WhenAccesingExistingChannel_WithNewOptions_ShouldUpdateExistingChannelWithNewOptions()
```
Comment (mattheworiordan) [inline src/IO.Ably.Tests/Rest/ChannelSpecs.cs]: Ha, ignore my last comment 👍

**Comment by mattheworiordan** [comment]
Great, very readable 👍

File: src/IO.Ably.Tests/Rest/ChannelSpecs.cs (hunk 1 above)
Comment (marto83) [inline src/IO.Ably.Tests/Rest/ChannelSpecs.cs]: Have a look at RSN3c -> 10 lines down :wink:

**Pull request merged**